src/simd/allocator.hpp

```diff
@@ -0,0 +1,60 @@
+//////////////////////////////////////////////////////////////////////////////////////
+// This file is distributed under the University of Illinois/NCSA Open Source License.
+// See LICENSE file in top directory for details.
+//
+// Copyright (c) 2016 Jeongnim Kim and QMCPACK developers.
+//
+// File developed by: 
+//
+// File created by: Jeongnim Kim, jeongnim.kim@intel.com, Intel Corp.
+//////////////////////////////////////////////////////////////////////////////////////
+// -*- C++ -*-
+/** @file allocator.hpp
+ */
+#ifndef QMCPLUSPLUS_ALLOCATOR_H
+#define QMCPLUSPLUS_ALLOCATOR_H
+
+#include <config.h>
+#include <vector>
+#include <cstdlib>
+
+#if (__cplusplus >= 201103L)
+#if defined(__INTEL_COMPILER)
+ #include <tbb/cache_aligned_allocator.h>
+#else
+ #include "simd/Mallocator.hpp"
+#endif
+
+namespace qmcplusplus
+{
+  template<class T>
+#if defined(__INTEL_COMPILER)
+    using aligned_allocator=tbb::cache_aligned_allocator<T>;
+#elif __bgq__
+    using aligned_allocator=std::allocator<T>;
```

Comment:
Mallocator is intended for machines having no aligned_allocator. BGQ std::allocator is aligned by default.